src/components/director-view/director-view.jsx

```diff
@@ -1,33 +1,80 @@
-import React from "react";
-import { useSelector } from "react-redux";
-import { useParams } from "react-router";
+import React, { useState, useEffect } from "react";
+import { Container, Row, Col } from "react-bootstrap";
+import { useParams, useNavigate } from "react-router-dom";
+import PropTypes from "prop-types";
 
+export const DirectorView = ({ token }) => {
+  const { directorName } = useParams();
+  const navigate = useNavigate();
+  const [director, setDirector] = useState(null);
 
-export const DirectorView = () => {
-  const params = useParams();
-  const directors = useSelector((state) => state.directors.list);
-  const director = directors.find((director) => director.name === params.name);
-
+  useEffect(() => {
+    if (!token) return;
+    (async () => {
+      try {
+        const response = await fetch(
+          `https://movieapi-aeueoes-projects.vercel.app/directors/${directorName}`,
+          {
+            headers: {
+              Authorization: `Bearer ${token}`,
+            },
+          }
+        );
+        const data = await response.json();
+        setDirector(data);
+      } catch (error) {
+        console.error("Error fetching director:", error);
+      }
+    })();
+  }, [token, directorName]);
   if (!director) {
-    return <div>Director not found!</div>;
+    return <div>Loading...</div>; // or a loading spinner, etc.
   }
 
   return (
-    <div className="director-view">
-      <h1>{director.name}</h1>
-      <p>
-        <strong>Bio:</strong> {director.bio}
-      </p>
-      <p>
-        <strong>Movies:</strong>
-        <ul>
-          {director.movies.map((movie) => (
-            <li key={movie.title}>
-              <Link to={`/movies/${movie.title}`}>{movie.title}</Link>
-            </li>
-          ))}
-        </ul>
-      </p>
-    </div>
+    <Container className="director-view">
+      <Row>
+        <Col md={8}>
+          <h1>{director.name}</h1>
+          <p>
+            <strong>Bio:</strong> {director.bio}
+          </p>
+          <p>
+            <strong>Birth:</strong> {director.birth}
+          </p>
+          {director.death && director.death !== "0" && (
+            <p>
+              <strong>Death:</strong> {director.death}
+            </p>
+          )}
+          <p>
+            <strong>Country:</strong> {director.country}
+          </p>
+          <p>
+            <strong>Other Films:</strong>
+            <ul>
+              {director.otherFilms &&
```

Comment:
You may want to refer otherFilms like this to avoid onload errors

director?.otherFilms && ...